modules/react-components/src/components/file-picker/file-picker.tsx

```diff
@@ -535,6 +559,87 @@ export const FilePicker: FC<FilePickerProps> = (props: FilePickerPropsAlias): Re
+    };
+
+    const DeleteFileConfirmationModal: ReactElement = (
+        <ConfirmationModal
+            onClose={ (): void => setShowFileDeleteConfirmation(false) }
+            onSecondaryActionClick={ (): void => setShowFileDeleteConfirmation(false) }
+            primaryActionLoading={ ongoingDeleteRequest }
+            onPrimaryActionClick={ deleteFile }
+            open={ showFileDeleteConfirmation }
+            type="negative"
+            primaryAction={ t("applications:confirmations.certificateDelete.primaryAction") }
+            secondaryAction={ t("applications:confirmations" +
+                ".certificateDelete.secondaryAction") }
+            data-testid={ "delete-confirmation-modal" }
+            closeOnDimmerClick={ false }>
+            <ConfirmationModal.Header
+                data-testid={ "delete-confirmation-modal-header" }>
+                { t("applications:confirmations.certificateDelete.header") }
+            </ConfirmationModal.Header>
+            <ConfirmationModal.Message
+                attached
+                negative
+                data-testid={ "delete-confirmation-modal-message" }>
+                { t("applications:confirmations.certificateDelete.message") }
+            </ConfirmationModal.Message>
+        </ConfirmationModal>
+    );
+
+    if (initialFile && showFileAsList) {
+
+        return (
+            <React.Fragment>
+                <Grid>
+                    <Grid.Row>
+                        <Grid.Column width={ 16 }>
+                            <Divider hidden/>
+                            <ResourceList
+                                fill
+                                relaxed={ false }
+                                className="file-list"
+                            >
+                                <ResourceListItem
+                                    itemHeader={ fileDisplayName }
+                                    avatar={ createFileAvatar(fileDisplayName) }
+                                    actionsColumnWidth={ 3 }
+                                    descriptionColumnWidth={ 9 }
+                                    actionsFloated="right"
+                                    actions={ [
+                                        {
+                                            "data-testid": "uploaded-file-delete-button",
+                                            icon: "trash alternate",
+                                            onClick: () => setShowFileDeleteConfirmation(true),
+                                            popupText: t("users:usersList.list.iconPopups.delete"),
+                                            type: "button"
+                                        }
+                                    ] }
+                                />
+                            </ResourceList>
+                        </Grid.Column>
+                    </Grid.Row>
+                </Grid>
+                { DeleteFileConfirmationModal }
```

Comment:
Is this intentional? hope we need the following here.

```suggestion
                <DeleteFileConfirmationModal />
```